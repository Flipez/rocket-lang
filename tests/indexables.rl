s = "abcdef"
puts(s[2])
puts(s[-2])
puts(s[:2])
puts(s[:-2])
puts(s[2:])
puts(s[-2:])
puts(s[1:-2])

s[2] = "C"
s[-2] = "E"
puts(s)

a = [1, 2, 3, 4, 5]
puts(a[2])
puts(a[-2])
puts(a[:2])
puts(a[:-2])
puts(a[2:])
puts(a[-2:])
puts(a[1:-2])

a[2] = 8
a[-2] = 9
puts(a)

h = {"a": 1, 2: true}
puts(h["a"])
puts(h[2])
h["a"] = 3
h["b"] = "moo"
puts(h["a"])
puts(h["b"])
puts(h[2])
