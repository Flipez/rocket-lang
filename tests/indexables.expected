"c"
"e"
"ab"
"abcd"
"cdef"
"ef"
"bcd"
"abCdEf"
3
4
[1, 2]
[1, 2, 3]
[3, 4, 5]
[4, 5]
[2, 3]
[1, 2, 8, 9, 5]
1
true
3
"moo"
true
